extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -240,3 +248,117 @@ func RestoreRKE2K3SSnapshot(client *rancher.Client, clusterName string, snapshot
 
 	return nil
 }
+
+// RetentionLimitCheckRKE1 is a check that validates that the number of automatic snapshots on the cluster is under the retention limit
+func RetentionLimitCheckRKE1(client *rancher.Client, clusterName string) error {
+	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
+	if err != nil {
+		return err
+	}
+
+	clusterResp, err := client.Management.Cluster.ByID(clusterID)
+	if err != nil {
+		return err
+	}
+
+	retentionLimit := clusterResp.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig.Retention
+
+	var s3Enabled bool = false
+	s3Config := clusterResp.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig.S3BackupConfig
+	if s3Config != nil {
+		s3Enabled = true
+	}
+
+	existingSnapshots, err := GetRKE1Snapshots(client, clusterName)
+	automaticSnapshots := []management.EtcdBackup{}
+	for _, snapshot := range existingSnapshots {
+		if !snapshot.Manual {
+			automaticSnapshots = append(automaticSnapshots, snapshot)
+		}
+	}
+
+	clusterNodes, err := nodes.GetNodes(client, clusterID, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	expectedSnapshotsNum := len(clusterNodes) * int(retentionLimit)
+	if s3Enabled {
+		expectedSnapshotsNum += int(retentionLimit)
+	}
+
+	if len(automaticSnapshots) > expectedSnapshotsNum {
+		msg := fmt.Sprintf("retention limit exceeded: expected %d snapshots, found %d snapshots", expectedSnapshotsNum, len(automaticSnapshots))
+		logrus.Info(automaticSnapshots)
+		return errors.New(msg)
+	}
+
+	logrus.Infof("Snapshot retention limit respected, Snapshots Expected: %v Snapshots Found: %v", expectedSnapshotsNum, len(automaticSnapshots))
+
+	return nil
+}
+
+// RetentionLimitCheckRKE2K3S is a check that validates that the number of automatic snapshots on the cluster is under the retention limit
+func RetentionLimitCheckRKE2K3S(client *rancher.Client, clusterName string) error {
```

Comment:
nit - similar to the RKE1 function, can this and the comment be renamed to something like `v2ProvRetentionLimitCheck`?